src/eth/TaraClient.sol

```diff
@@ -158,13 +146,24 @@ contract TaraClient is IBridgeLightClient {
     }
 
     /**
-     * @dev Finalizes a block by verifying the signatures and processing the changes.
-     * @param b PillarBlockWithChanges.
-     * @param signatures An array of Signature structs representing the signatures of the block.
+     * @dev Finalizes blocks by verifying the signatures for the last blocks
+     * @param encoded_blocks list if PillarBlockWithChanges.
+     * @param last_block_sigs An array of Signature structs representing the signatures of the last block.
      */
-
-    function finalizeBlock(bytes memory b, CompactSignature[] memory signatures) public {
-        _finalizeBlock(PillarBlock.fromBytes(b), PillarBlock.getHash(b), signatures);
+    function finalizeBlocks(bytes[] memory encoded_blocks, CompactSignature[] memory last_block_sigs) public {
```

Comment:
```suggestion
    function finalizeBlocks(bytes[] memory encodedBlocks, CompactSignature[] memory lastBlockSigs) public {
```
If we wanna keep constant let's use either `snake_case` or `camelCase`

Reply:
why do we send pillar block as bytes, but vote as CompactSignature struct ?

Reply:
Because bytes are used to get the hash of the block. So it feels to me like it will be little bit cheaper to have this bytes for the hash and decode a block than to accept a block and encode it to bytes. But maybe it is not the case. @Elod23 any idea?

Reply:
I highly suggest to keep sending structs in places possible. I think this is the optimal way here.

Reply:
any easy way how to mease abi.encode vs abi.decode gas ? Because if it is the same or similar, I would vote for using struct also for Pillar block. It would be cleaner...